File: LeetCode/2351. First Letter to Appear Twice/solution.cpp
/**
 * Runtime: 0 ms
 * Memory Usage: 6 MB 
 */
class Solution {
public:
    char repeatedCharacter(string s) {
        unordered_map<char, int> cnt;
        for (auto &c: s) {
            cnt[c]++;
            if (cnt[c] == 2) {
                return c;
            }
        }
        return {};
    }
};

File: LeetCode/2351. First Letter to Appear Twice/solution.py
"""
Runtime: 36 ms
Memory Usage: 14.9 MB
"""
class Solution:
    def repeatedCharacter(self, s: str) -> str:
        a = ord('a')
        alphabet = [0] * 26
        for c in s:
            alphabet[ord(c) - a] += 1
            if alphabet[ord(c) - a] == 2:
                return c
        return ''
